### ISSUE TITLE
[Payments] Epic: Module Payments

### ISSUE THREAD

**Issue by naui95** [InvoicePlane/InvoicePlane-v2#17]
- [ ] #32

**Comment by nielsdrost7** [comment]
## SMART Plan — Payments module epic status

> **Context/Epic**: Payments
>
> This is an epic/tracking issue, not a single feature — it aggregates the sub-issues below. The Measurable section tracks sub-issue completion rather than fabricating tests for the epic itself; each sub-issue gets its own SMART Plan when it's processed individually.

### Specific
Track delivery of the Payments module: manual payment CRUD (done) plus planned expansions — inline payment entry from the invoice list, payable-only invoice filtering, an online payments journal, gateway integrations (BTCPay, bank-feed retrieval), and permission enforcement.

### Measurable
- [x] `PaymentResource` (list/create/edit/delete) with full CRUD test coverage in `Modules/Payments/Tests/Feature/PaymentsTest.php`.
- [ ] #32 — Enter payment inline from the invoice list — not started; no such action exists in `InvoicesTable`.
- [ ] #201 — Enter payment modal triggered from the invoice list — not started; same gap as #32, almost certainly the same piece of work.
- [ ] #225 — Invoice selector filters to payable invoices only — not started; `PaymentForm`'s invoice `Select` has no status filter today.
- [ ] #144 — Online Payments Journal — not started; no `GatewayTransaction` model/table exists, and `PaymentObserver` is fully commented out, so there's nothing to log regardless.
- [ ] #373 — BTCPay Server integration — not started; also blocked on a public/guest invoice view that doesn't exist anywhere in the codebase yet.
- [ ] #378 — SMTP OAuth for payment notification emails — not started (no OAuth mail driver configuration found in `Modules/Core`).
- [ ] #495 — Permission enforcement for Payments — partially in place: `PaymentsTable` already gates each row action behind `auth()->user()?->can(Permission::...)` (edit/email_receipt/refund/delete), but there's no dedicated policy class or authorization test suite confirming coverage is exhaustive.

**Definition of done for this epic**: all sub-issues above are closed, `Modules/Payments/Tests/Feature/` covers each new capability, and at least one real (non-stub) gateway integration exists to populate the journal.

### Achievable
1. #225 (invoice filter) is the smallest, fully independent, unblocked piece of work.
2. #32/#201 (inline payment entry from the invoice list) describe the same UI entry point and should be sequenced together.
3. #144 (journal) can ship its table/UI shell independently, but stays functionally empty until a real gateway (#373 or another driver) writes to it — sequence a gateway alongside or before it if the acceptance tests need real data.
4. #373 (BTCPay) is blocked on a public/guest invoice view existing, which isn't currently scoped under any tracked sub-issue — flag this as a prerequisite that needs its own issue.
5. #495 (permission enforcement) should land last, once the new actions (#32/#201/#144/#373) exist and need guarding.

### Relevant
Payments is the terminal step of the invoice lifecycle; the CRUD basics work today, but the module has no live gateway, no audit trail for gateway activity, and rough edges in the manual-entry UX that create daily friction for staff.

### Time-bound
No fixed date — epic closes when its sub-issues close. Re-review at each sprint boundary.
